x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -429,11 +438,17 @@ public static void buildRoleFromDescriptors(
                     }
                 });
             }
+
+            if (descriptor.hasWorkflowsRestriction()) {
+                assert roleDescriptors.size() == 1 : "only single role descriptor with workflow restriction is allowed";
```

Comment:
I think this should be an exception to counter potential abuse of custom role provider